akka-typed/src/test/scala/docs/akka/typed/IntroSpec2.scala

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (C) 2014-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package docs.akka.typed
+
+//#imports
+import akka.typed._
+import akka.typed.scaladsl.Actor._
+import akka.typed.AskPattern._
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.concurrent.Await
+//#imports
+import akka.testkit.AkkaSpec
+import akka.typed.TypedSpec
+
+object IntroSpec2 {
+
+  //#hello-world-actor
+  object HelloWorld {
+    final case class Greet(whom: String, replyTo: ActorRef[Greeted])
+    final case class Greeted(whom: String)
+
+    // TODO Is the small difference between Stateless and Stateful worth separate classes?
+    //      Stateful returning Same is the same as Stateless
+    val greeter: Behavior[Greet] = Stateless { (_, msg) ⇒
+      println(s"Hello ${msg.whom}!")
+      msg.replyTo ! Greeted(msg.whom)
+    }
+  }
+  //#hello-world-actor
+
+  //#chatroom-actor
+  object ChatRoom {
+    //#chatroom-protocol
+    sealed trait Command
+    final case class GetSession(screenName: String, replyTo: ActorRef[SessionEvent])
+      extends Command
+    //#chatroom-protocol
+    //#chatroom-behavior
+    private final case class PostSessionMessage(screenName: String, message: String)
+      extends Command
+    //#chatroom-behavior
+    //#chatroom-protocol
+
+    sealed trait SessionEvent
+    final case class SessionGranted(handle: ActorRef[PostMessage]) extends SessionEvent
+    final case class SessionDenied(reason: String) extends SessionEvent
+    final case class MessagePosted(screenName: String, message: String) extends SessionEvent
+
+    final case class PostMessage(message: String)
+    //#chatroom-protocol
+    //#chatroom-behavior
+
+    val behavior: Behavior[GetSession] = {
+      var sessions = List.empty[ActorRef[SessionEvent]]
```

Comment:
Indeed, there is no way to make this (incorrect) usage impossible. The correct way would be to use the Deferred constructor and create the mutable state when the actor is created.